app/lib/fields/form_builder.rb

```diff
@@ -40,6 +40,12 @@ def output_html(field, options = {})
                   else
                     Fields::BuiltinField.new(opts)
                   end
-    input_field.input(self)
+
+    type = input_type(field.name.to_sym, input_field.builder_options)
```

Comment:
we need to do this here, and not earlier, because we need `:collection` to be set, because it will be used by Formtastic's `default_input_type` to detect a `:select`, https://github.com/3scale/porta/blob/0b2efa96557e619c558e3d3da471022d3cfa6527/app/lib/fields/base_field.rb#L45